node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -4304,6 +4304,12 @@ mod tests {
     #[test]
     fn standard_gossip_containing_non_transmitting_neighbor_node_that_accepts_connections_but_has_no_node_addr_is_accepted(
     ) {
+        // src_root, which accepts connections, is about to tell dest_root about non_malefactor,
+        // which also accepts connections. The NodeRecord for non_malefactor will not contain its
+        // NodeAddr. It's a violation for src_root to claim to accept connections but not to
+        // provide its own NodeAddr, but this test ensures that it's okay for src_root not to
+        // provide the NodeAddr of non_malefactor, since non_malefactor is not the transmitter of
+        // this Gossip.
```

Comment:
Oh thanks, this comment makes a lot of sense. It's bulky but I hope we can afford such a thing from time to time.